ghcide/src/Development/IDE/GHC/Warnings.hs

```diff
@@ -33,15 +33,12 @@ withWarnings diagSource action = do
   warnings <- newVar []
   let newAction :: DynFlags -> LogActionCompat
       newAction dynFlags logFlags wr _ loc prUnqual msg = do
-        let wr_d = map ((wr,) . third3 (attachReason wr)) $ diagFromErrMsg diagSource dynFlags $ mkWarnMsg dynFlags wr logFlags loc prUnqual msg
+        let wr_d = map ((wr,) . modifyFdLspDiagnostic (attachReason wr)) $ diagFromSDocErrMsg diagSource dynFlags (mkWarnMsg dynFlags wr logFlags loc prUnqual msg)
```

Comment:
Note that `attachReason` arguably does the wrong thing here: it prints the warning _flag_ into the code, which seems wrong.

Reply:
That was a source of a bit of confusion when I got started testing my changes, I forgot that it's technically a bug - I've dropped it in 4fed9877, I doubt anyone relied on this.

That said, there are a few other places where I see the `_code` field being used weirdly.

Reply:
(In the meantime, reverted)

Reply:
Looking deeper, `attachReason` was introduced in f03a7fa55d909c864771f0989abace8793b5b15a, explicitly for code actions that would disable a warning.

If we reimplement the code action in some other way, then we should be good.

Reply:
Yes, that all sounds right. Ideally we'd have the structured diagnostics and then we'd be able to do this directly, in the mean time using `_data` seems better.

Reply:
Alright, I've solved this in 7bb82d0b